library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -150,7 +150,7 @@ public void onLocationChanged(Location location) {
         // Set waiting as false because we got at least one, even though we keep tracking user's location
         setWaiting(false);
 
-        if (!getConfiguration().keepTracking()) {
+        if (!getConfiguration().keepTracking() && getSourceProvider().isGoogleApiClientConnected()) {
```

Comment:
The only necessary part is this, maybe you can consider to create another PR with only this line changed?
And of course, you should update the test 👍